TheWidlarzGroup/react-native-video
# feat: add typescript

## Summary
This PR add built-in types and configuration for typescript. It is also step to prepare for new architecture support (codegen). Changes and types are based on [New Arch PR](https://github.com/react-native-video/react-native-video/pull/3204) by @yungblud 

## Why is needed ?
With built-in types we have up to date types on every version and we don't need to install an rely on additional package for types `@types/react-native-video`. Overall built-in types **increase the comfort of both developers and maintainers**.
But `@types/react-native-video` **should stay** to support older versions of `react-native-video`

## How to test ?
1. Checkout for branch of PR
2. Install dependencies
3. Run `yarn build`
4. update dependencies in (basic) example app
5. Check if types are correctly imported
6. Example app should build and run
7. Types for `Video` component should be available
